packages/components/addon/components/hds/tabs/index.js

```diff
@@ -1,11 +1,74 @@
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
 export default class HdsTabsIndexComponent extends Component {
-  // UNCOMMENT THIS IF YOU NEED A CONSTRUCTOR
-  // constructor() {
-  //   super(...arguments);
-  //   // ADD YOUR ASSERTIONS HERE
-  // }
+  @tracked tabNodes = [];
+  @tracked tabIds = [];
+  @tracked panelNodes = [];
+  @tracked panelIds = [];
+  @tracked selectedTabIndex;
+
+  @action
+  didInsert() {
+    // default starting tab index
+    let initialTabIndex = 0;
+
+    this.tabNodes.forEach((tabElement, index) => {
+      if (tabElement.hasAttribute('data-is-selected')) {
+        initialTabIndex = index;
+      }
+    });
+    this.selectedTabIndex = initialTabIndex;
+  }
+
+  @action
+  didInsertTab(element) {
+    this.tabNodes = [...this.tabNodes, element];
+    this.tabIds = [...this.tabIds, element.id];
+  }
+
+  @action
+  didInsertPanel(panelId, element) {
+    this.panelNodes = [...this.panelNodes, element];
+    this.panelIds = [...this.panelIds, panelId];
+  }
+
+  @action
+  handleClick(tabIndex, e) {
+    e.preventDefault();
+    this.selectedTabIndex = tabIndex;
+  }
+
+  @action
+  handleKeyUp(tabIndex, e) {
+    const leftArrow = 37;
```

Comment:
[question] do you know if these re the same across all different types of keyboards? I was using a french keyboard the other day and the keys were definitely in different places.

Reply:
It's using the ASCII code which should be standard for the arrow keys across various keyboards and languages. 

I found this: https://stackoverflow.com/questions/20863329/are-up-down-left-right-arrow-keycodes-always-the-same